src/s3fetch/command.py

```diff
@@ -291,6 +316,19 @@ def _download_object(self, key: str, exit_event: threading.Event) -> None:
                     Callback=self._download_callback,
                     Config=s3transfer_config,
                 )
+                if on_download is not None:
+                    try:
```

Comment:
Don't think we should be allowing the `on_download` callback to silently fail, really we should get rid of the `try/except` and throw the exception so the user knows things are not working as intended. They can ignore the exception in their code if they want to.